### PR TITLE
API side error code changes

### PR DESCRIPTION
**Pull request by atelesman**
To test: Pull WAY-81 from the API side and run it. Pull WAY-81 from app side as well, and change server address in displaymapactivity to your ip address. 
To test "no route exists" message: I chose the first choice on the list the "Buckeye Village student housing unit [...]" and Bolz hall. A message saying no route exists between the two nodes should pop up.
To test "no start error" message: I did this when I was off campus and chose an oncampus spot. A message saying no start node found should appear.

### CODE REVIEW

File: src/main/java/com/RouteController.java [hunk 1]
```diff
@@ -96,7 +98,7 @@ public Route generateRoute(@RequestParam(value = "dest") int destID, @RequestPar
 
         if (startNode == null)
         {
-            return null; // Starting node could not be found within 300 feet of current location. Route not available.
+            return new Route(null, null, null, "A starting node could not be found for your current location."); // Starting node could not be found within 300 feet of current location. Route not available.
```
Comment (JDSustar) [inline src/main/java/com/RouteController.java]: I'm not sure this error message is really user friendly.

Something like "You are not close enough to campus or paths on campus to calculate a route using your current location"

File: src/main/java/com/RouteController.java [hunk 2]
```diff
@@ -114,14 +116,13 @@ public Route generateRoute(@RequestParam(value = "dest") int destID, @RequestPar
...
-            Route currentRoute = new Route(currentRouteNodes, new Door(-1, "Current Location", new Coordinate(currLat, currLong, Coordinate.TYPE.GCS)), endDoor);
+            Route currentRoute = new Route(currentRouteNodes, new Door(-1, "Current Location", new Coordinate(currLat, currLong, Coordinate.TYPE.GCS)), endDoor, null);
 
             if(bestRoute == null || currentRoute.getLengthInFeet() < bestRoute.getLengthInFeet())
             {
                 bestRoute = currentRoute;
             }
         }
-
         return bestRoute;
```
Comment (JDSustar) [inline src/main/java/com/RouteController.java:129]: You do a null check in the previous method, but not this one. I would suggest doing it here, too.

File: src/main/java/com/Route.java [hunk 3]
```diff
@@ -9,19 +9,21 @@
     private Door endDoor;
     private double lengthInFeet;
     private final List<Node> route;
+    private String errorMsg;
 
-    public Route(List<Node> route, Door start, Door end){
+    public Route(List<Node> route, Door start, Door end, String errorMsg){
```
Comment (JDSustar) [inline src/main/java/com/Route.java]: This is fine, but in the future, considering something like keeping this method intact and making a new constructor.

```
public Route(List<Node> route, Door start, Door end){ ... }

public Route(List<Node> route, Door start, Door end, String errorMsg)
{
    this(route, start, end);
    this.errorMsg = errorMsg;
}
```

This results in only having to change the use of the constructor when you want to add an error message. All of the other ones don't have the dangling `null`

**Comment by atelesman** [comment]
Great insight, Jason, thanks. I have pushed the changes.

**Comment by JDSustar** [comment]
The changes look so much cleaner now. Good fixes. :smile: :+1:

**Pull request merged**